vignettes/articles/zip_zap_models.Rmd

```diff
@@ -160,28 +179,35 @@ where $p=\text{logit}^{-1}(\theta)$.  We achieve the truncation of zeros in the
 Here ZIP distribution only governs the positive counts and the absences will be handled by a separate binomial model which can have its own covariates. Note, that we exclude the observations with absent nests from the ZIP model by subsetting the data to include only observations where `present` is `TRUE`.
 
 ```{r fig-pred-zap, fig.cap="Predictions from zero-adjusted model",fig.width=7, fig.height=5, out.width="80%", fig.align='center'}
-fit_zap <- bru(comps,
-               like(family = "binomial", data=counts_df,
-                    formula = present~vegetation+field+Intercept,
-                   domain = list(geometry = px_mesh)),
-               like(family ="zeroinflatedpoisson0",
-                    data=counts_df[counts_df$present,],
-                    formula = count~elevation+field+Intercept,
-                    E=area,
-                    domain = list(geometry = px_mesh),
-                    control.family = list(hyper=list(theta=list(initial=-20, fixed=TRUE)))))
+fit_zap <- bru(
+  comps,
+  like(
+    family = "binomial", data = counts_df,
+    formula = present ~ vegetation + field + Intercept,
+    domain = list(geometry = px_mesh)
+  ),
+  like(
+    family = "zeroinflatedpoisson0",
+    data = counts_df[counts_df$present, ],
+    formula = count ~ elevation + field + Intercept,
+    E = area,
+    domain = list(geometry = px_mesh),
```

Comment:
No need for domain; not a "cp" model.

Reply:
I included the size of the cell as `E`. There's a minor irregularity in the size of the pixel due to projection. I kept it here for illustrative purposes, but as long as projection is not too crazy, I guess it could be dropped as well (i.e assumed uniform)?

Reply:
I am also not sure if it is a good idea to subset the data to only the `present` observation in ZIP part of the model, since the prior for `theta` is fixed. But judging by WAICC it is a necessary thing. Not sure I understand why.

Reply:
Including the cell area as E seems like the appropriate thing to do. No need to assume equal areas.

Subsetting the data like that for the ZIP part is essential in the hurdle construction; the "p \approx 0" is an implementation hack, and you don't want any `\log(\epsilon)` to appear in the computations; that would just add a constant large negative number (proportional to the number of zeros) to the log-pmf.